### DISCUSSION TITLE
时不时出现异常信息：[【Exception】：The transaction operation cannot be performed because there are pending requests working on this transaction.]

### DISCUSSION THREAD

**Discussion by km93522719** [dotnetcore/FreeSql#1236] [General]
using IUnitOfWork unitOfWork = _unitOfWorkManager.Begin(isolationLevel: IsolationLevel.Snapshot);
try
{
   //业务相关的代码
    ...
    unitOfWork.Commit();
}
catch (Exception e)
{
    unitOfWork.Rollback();
    throw new CustomException(ExceptionConst.FlowSubmitFail);
}
时不时出现异常信息：[【Exception】：The transaction operation cannot be performed because there are pending requests working on this transaction.]

**Comment by 2881099** [comment]
检查一下异步，是不是没有 await

**Reply by km93522719** [reply]
> 检查一下异步，是不是没有 await

检查过都添加了await，这个问题有点恑异，大部分都是正常，就是偶尔出现一下
